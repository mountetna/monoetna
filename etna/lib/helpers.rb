require_relative './etna/clients'
require_relative './etna/environment_scoped'

module WithEtnaClients
  def environment
    EtnaApp.instance.environment
  end

  def token
    if !ENV['TOKEN']
      puts "No environment variable TOKEN is set. Set your token with `export TOKEN=<your.janus.token>` before running Etna commands."
      exit
    end
    ENV['TOKEN']
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(token: token, host: EtnaApp.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(token: token, host: EtnaApp.instance.config(:metis, environment)[:host])
  end

  def janus_client
    @janus_client ||= Etna::Clients::Janus.new(token: token, host: EtnaApp.instance.config(:janus, environment)[:host])
  end
end

module WithLogger
  def logger
    EtnaApp.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
