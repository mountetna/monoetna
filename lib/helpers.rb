require_relative './etna/clients'
require_relative './etna/environment_scoped'

module WithEtnaClients
  def environment
    EtnaApp.instance.environment
  end

  def exit(status=true)
    WithEtnaClients.exit(status)
  end

  # Abstraction used to prevent accidental exist in specs.
  def self.exit(status)
    Kernel.exit(status)
  end

  def token(ignore_environment: false)
    unless ignore_environment
      if environment == :many
        raise "You have multiple environments configured, please specify your environment by adding --environment <staging|production|development>"
      elsif environment == :none
        raise "You do not have a successfully configured environment, please run #{program_name} config set https://polyphemus.ucsf.edu"
      end
    end

    env_token = ENV['TOKEN']
    if !env_token
      puts "No environment variable TOKEN is set.  You should set your token with `export TOKEN=<your.janus.token>` before running."
      redirect = EtnaApp.instance.config(:auth_redirect)

      if redirect.nil? && EtnaApp.instance.environment == :production
        redirect = 'https://janus.ucsf.edu/'
      end

      unless redirect.nil?
        puts "Open your browser to #{redirect} to complete login and copy your token."
      end

      exit
    end

    env_token
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:magma, environment) || {})
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:metis, environment) || {})
  end

  def janus_client
    @janus_client ||= Etna::Clients::Janus.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:janus, environment) || {})
  end

  def polyphemus_client
    @polyphemus_client ||= Etna::Clients::Polyphemus.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:polyphemus, environment) || {})
  end
end

module WithLogger
  def logger
    EtnaApp.instance.logger
  end
end

module StrongConfirmation
  def confirm
    puts "Confirm Y/n:"
    input = STDIN.gets.chomp
    if input.downcase != "y"
      puts "Bailing..."
      return false
    end

    true
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
