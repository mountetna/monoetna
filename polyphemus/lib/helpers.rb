require 'etna/clients'
require 'etna/environment_scoped'
require 'shellwords'

module WithSlackNotifications
  def notify_slack(message, messenger:self.class.name, channel:)
    `/bin/post-to-slack.sh #{Shellwords.escape(messenger)} #{Shellwords.escape(channel)} #{Shellwords.escape(message)} || true`
  end
end

module WithEtnaClients
  def environment
    Polyphemus.instance.environment
  end

  def token
    Polyphemus.instance.config(:polyphemus, environment)[:token]
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
      token: token,
      host: Polyphemus.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
      token: token,
      host: Polyphemus.instance.config(:metis, environment)[:host])
  end

  def metis_client=(client)
    @metis_client = client if Polyphemus.instance.test?
  end
end

module WithLogger
  def logger
    Polyphemus.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
