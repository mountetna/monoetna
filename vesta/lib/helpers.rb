require 'etna/clients'
require 'etna/environment_scoped'
require 'shellwords'

module WithEtnaClients
  def environment
    Vesta.instance.environment
  end

  def token
    Vesta.instance.config(:vesta, environment)[:token]
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
      token: token,
      host: Vesta.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
      token: token,
      host: Vesta.instance.config(:metis, environment)[:host])
  end

  def metis_client=(client)
    @metis_client = client if Vesta.instance.test?
  end

  def janus_client
    @magma_client ||= Etna::Clients::Janus.new(
      token: token,
      host: Vesta.instance.config(:janus, environment)[:host])
  end
end

module WithLogger
  def logger
    Vesta.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
