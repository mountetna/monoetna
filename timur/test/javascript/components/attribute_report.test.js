import React from 'react';
import {Provider} from 'react-redux';
import {mockStore} from '../helpers';
import {render, screen, waitFor, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import AttributeReport from '../../../lib/client/jsx/components/model_map/attribute_report';

const monster = require('../fixtures/template_monster.json');

describe('AttributeReport', () => {
  let store = mockStore({});

  it('renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AttributeReport attribute={monster.attributes.name} />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with model action buttons for admin user', async () => {
    store = mockStore({
      magma: {
        models: {
          monster: require('../fixtures/template_monster.json')
        }
      },
      janus: {projects: require('../fixtures/project_names.json')},
      user: {
        permissions: {
          labors: {
            role: 'administrator'
          }
        }
      }
    });

    const {asFragment} = render(
      <Provider store={store}>
        <AttributeReport attribute={monster.attributes.name} />
      </Provider>
    );

    await waitFor(() => screen.getByText('Edit'));

    expect(screen.queryByText('Remove')).toBeFalsy();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with remove attribute buttons for admin user, removable attribute', async () => {
    store = mockStore({
      magma: {
        models: {
          monster: require('../fixtures/template_monster.json')
        }
      },
      janus: {projects: require('../fixtures/project_names.json')},
      user: {
        permissions: {
          labors: {
            role: 'administrator'
          }
        }
      }
    });

    const {asFragment} = render(
      <Provider store={store}>
        <AttributeReport attribute={monster.attributes.species} />
      </Provider>
    );

    await waitFor(() => screen.getByText('Remove'));

    expect(screen.getByText('Edit')).toBeTruthy();
    expect(asFragment()).toMatchSnapshot();
  });
});
