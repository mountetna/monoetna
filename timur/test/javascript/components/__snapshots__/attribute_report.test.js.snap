// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeReport renders 1`] = `
<div
  className="MuiGrid-root makeStyles-attribute_report-2"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-attribute_card-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root makeStyles-heading-8 undefined MuiGrid-container"
    >
      <h6
        className="MuiTypography-root makeStyles-name-9 MuiTypography-h6"
      >
        Attribute
      </h6>
       
      <h6
        className="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
      >
        name
      </h6>
      <div
        className="MuiGrid-root MuiGrid-container"
        style={
          Object {
            "alignItems": "end",
            "flex": 1,
            "justifyContent": "right",
            "width": "auto",
          }
        }
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textSecondary MuiButton-textSizeSmall MuiButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Show data sample"
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Sample
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      className="MuiCardContent-root makeStyles-content-4"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
        >
          attribute_type
        </div>
        <div
          className="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
        >
          identifier
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
        >
          display_name
        </div>
        <div
          className="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
        >
          Name
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AttributeReport renders with remove attribute buttons for admin user, removable attribute 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-attribute_report-2"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-attribute_card-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiGrid-root makeStyles-heading-8 undefined MuiGrid-container"
      >
        <h6
          class="MuiTypography-root makeStyles-name-9 MuiTypography-h6"
        >
          Attribute
        </h6>
         
        <h6
          class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
        >
          species
        </h6>
        <div
          class="MuiGrid-root MuiGrid-container"
          style="width: auto; flex: 1; align-items: end; justify-content: right;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            title="Remove Attribute"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              Remove
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textSecondary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            title="Edit Attribute"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </span>
              Edit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textSecondary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            title="Show data sample"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Sample
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiCardContent-root makeStyles-content-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            attribute_type
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            string
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            display_name
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            Species
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            validation_type
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            Regexp
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            validation
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            "^[a-z\\\\s]+$"
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AttributeReport renders with remove link button for admin user, collection (link) attribute 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-attribute_report-2"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-attribute_card-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiGrid-root makeStyles-heading-8 undefined MuiGrid-container"
      >
        <h6
          class="MuiTypography-root makeStyles-name-9 MuiTypography-h6"
        >
          Attribute
        </h6>
         
        <h6
          class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
        >
          monster
        </h6>
        <div
          class="MuiGrid-root MuiGrid-container"
          style="width: auto; flex: 1; align-items: end; justify-content: right;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            title="Remove Link"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              Remove Link
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiCardContent-root makeStyles-content-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            attribute_type
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            collection
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            link_model_name
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            monster
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            display_name
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            Monster
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AttributeReport renders with remove link button for admin user, link attribute 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-attribute_report-2"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-attribute_card-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiGrid-root makeStyles-heading-8 undefined MuiGrid-container"
      >
        <h6
          class="MuiTypography-root makeStyles-name-9 MuiTypography-h6"
        >
          Attribute
        </h6>
         
        <h6
          class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
        >
          habitat
        </h6>
        <div
          class="MuiGrid-root MuiGrid-container"
          style="width: auto; flex: 1; align-items: end; justify-content: right;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            title="Remove Link"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              Remove Link
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiCardContent-root makeStyles-content-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            attribute_type
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            link
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            link_model_name
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            habitat
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            display_name
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            Habitat
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AttributeReport renders without model action buttons for admin user, non-editable attribute 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-attribute_report-2"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-attribute_card-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiGrid-root makeStyles-heading-8 undefined MuiGrid-container"
      >
        <h6
          class="MuiTypography-root makeStyles-name-9 MuiTypography-h6"
        >
          Attribute
        </h6>
         
        <h6
          class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
        >
          name
        </h6>
        <div
          class="MuiGrid-root MuiGrid-container"
          style="width: auto; flex: 1; align-items: end; justify-content: right;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textSecondary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            title="Show data sample"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Sample
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiCardContent-root makeStyles-content-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            attribute_type
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            identifier
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            display_name
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            Name
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AttributeReport renders without model action buttons for editor user 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-attribute_report-2"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-attribute_card-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiGrid-root makeStyles-heading-8 undefined MuiGrid-container"
      >
        <h6
          class="MuiTypography-root makeStyles-name-9 MuiTypography-h6"
        >
          Attribute
        </h6>
         
        <h6
          class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
        >
          species
        </h6>
        <div
          class="MuiGrid-root MuiGrid-container"
          style="width: auto; flex: 1; align-items: end; justify-content: right;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textSecondary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            title="Show data sample"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Sample
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiCardContent-root makeStyles-content-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            attribute_type
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            string
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            display_name
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            Species
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            validation_type
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            Regexp
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root makeStyles-type-5 MuiGrid-item MuiGrid-grid-xs-3"
          >
            validation
          </div>
          <div
            class="MuiGrid-root makeStyles-value-6 MuiGrid-item MuiGrid-grid-xs-9"
          >
            "^[a-z\\\\s]+$"
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
